### PR TITLE
Add file example formatting to code blocks

### LINKED ISSUE #134: Label file reference code blocks with title
## Proposed change

Update the code blocks that reference a file with formatting that adds a label to the code block for better clarity. Easier to see it's a file, and not a command/output

**example**
Screenshot from 2019-03-08 21-24-11

To accomplish this, we would add the following markdown to the file code blocks:
```
!!! example "secrets.yaml"
    ```yaml
    apiVersion: bitnami.com/v1alpha1
    kind: SealedSecret
    metadata:
      creationTimestamp: null
      name: alexellis-fn-secrets
      namespace: openfaas-fn
    spec:
      encryptedData:
        incoming-webhook-url: (redacted)
    ```
```
> Note: The code block is also indented so that the outline will wrap the entire block

## Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->
Want to make it easier to know that a block of code is reference to a file, and not some terminal input or output.


## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* FaaS-CLI version ( Full output from: `faas-cli version` ):
0.8.3
* Docker version `docker version` (e.g. Docker 17.0.05 ):
18.06.3-ce
* Are you using Docker Swarm or Kubernetes (FaaS-netes)?
Both :smile: 

* Operating System and version (e.g. Linux, Windows, MacOS):
Ubuntu 18.10

* Link to your project or a code example to reproduce issue:
Image included above
* Please also follow the [troubleshooting guide](https://github.com/openfaas/faas/blob/master/guide/troubleshooting.md) and paste in any other diagnostic information you have:
N/A

### PR DESCRIPTION
**Pull request by rgee0**
Signed-off-by: Richard Gee <richard@technologee.co.uk>

## Description
Initial sweep to add example file headings to code blocks as described in #134.
Standardises how files are described (removes early *filename.ext* notation)
Some smaller code blocks have been omitted at this stage.  These can be reviewed and, if necessary, picked up in a later change.

## Motivation and Context
- [x] I have raised an issue to propose this change (Fixes #134)

## How Has This Been Tested?
Review netlify preview link on raising of PR
https://deploy-preview-138--openfaas.netlify.com/conceptual/
image

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

### CODE REVIEW

File: docs/cli/build.md [hunk 1]
```diff
@@ -39,11 +39,12 @@ faas-cli build --lang python3 --build-option dev [--build-option debug]
 
 or in YAML:
 
-```yaml
-    build_options:
-    - debug
-    - dev
-```
+!!! example "stack.yml"
```
Comment (alexellis) [inline docs/cli/build.md]: New one on me. Is the indentation needed?

Comment (rgee0) [reply docs/cli/build.md]: Apparently.  Additionally it seems I may not have indented enough.

**Comment by burtonr** [comment]
This looks really good! It really makes a big difference in some of the "Reference" pages where the files stick out.

File: docs/cli/build.md (hunk 1 above)
Comment (alexellis) [reply docs/cli/build.md]: Merge or no?

I'm not keen on additional maintenance

**Pull request closed without merging**